mlflow/cli.py

```diff
@@ -145,49 +146,86 @@ def run(uri, entry_point, version, param_list, experiment_name, experiment_id, m
+                                 "store are SQLAlchemy compatible implementation or local storage."
+                                 " Supports various SQLAlchemy compatible database like SQLite,"
+                                 " MySQL, PostgreSQL. As an example MySQL backed store can be"
+                                 " configured using connection string."
+                                 "'mysql://<user_name>:<password>@<host>:<port>/<database_name>' "
+                                 "By default file based backed store"
+                                 " will be used. (default: ./mlruns).")(function)
+    function = click.option("--default-artifact-root", metavar="URI", default=None,
+                            help="Local or S3 URI to store artifacts, for new experiments. "
+                                 "Note that this flag does not impact already-created experiments. "
+                                 "Default: " + DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH)(function)
+    function = click.option("--host", "-h", metavar="HOST", default="127.0.0.1",
+                            help="The network address to listen on (default: 127.0.0.1). "
+                                 "Use 0.0.0.0 to bind to all addresses if you want to"
+                                 " access the tracking server from other machines.")(function)
+    function = click.option("--port", "-p", default=5000,
+                            help="The port to listen on (default: 5000).")(function)
+    function = click.option("--gunicorn-opts", default=None,
+                            help="Additional command line options forwarded to gunicorn"
+                                 " processes.")(function)
+    return function
 
-    The UI will be visible at http://localhost:5000 by default.
-    """
-    # Ensure that both backend_store_uri and default_artifact_uri are set correctly.
+
+def _run_tracking_server(backend_store_uri, default_artifact_root, host, port, workers,
+                         static_prefix, gunicorn_opts):
     if not backend_store_uri:
         backend_store_uri = DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH
 
     if not default_artifact_root:
         if _is_local_uri(backend_store_uri):
             default_artifact_root = backend_store_uri
         else:
-            default_artifact_root = DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH
+            eprint("Option 'default-artifact-root' is required, when backend store is not "
+                   "local file based.")
+            sys.exit(1)
+
+    # Validate the URIs
+    try:
+        _get_store(backend_store_uri)
+    except Exception as e:  # pylint: disable=broad-except
+        # We need a broad exception here to catch exceptions in optional
+        # dependencies like SQLAlchemy
+        eprint("Error related to option backend-store-uri '{}': {}".format(
+            backend_store_uri, e))
+        sys.exit(1)
+    try:
+        artifact_repo = get_artifact_repository(default_artifact_root)
```

Comment:
Are there use cases where a user may point to an artifact_root where the server doesn't have access to the artifact_root?

BTW, `get_artifact_repository` is supposed to take the artifact URI of a run, and `list_artifacts` takes the artifact path which is relative to the artifact URI of a run.

Reply:
It's possible in the cases like credential files with the wrong permissions, typos, and missing K8 volume mounts. To catch such issues, we'd want to instantiate a GCS/S3/Azure client, which occurs in `list_artifacts`

Reply:
I used `get_artifact_repository` in this way since the implementation only seems to use the scheme at the moment. `list_artifacts` also doesn't seem to raise an `MlflowException` if the URI is not a run.

This is the only somewhat maintainable way to check whether the backend store URI creds are correct. It might not make sense to do this check if the implementation of the above methods will change.

Reply:
Also, this isn't going to work with `dbfs:/` artifact URIs right? It might be easier to do this once we remove the `store` argument from `get_artifact_repository`. I believe @sueann might be working on something like this.

For this PR, can we remove the artifact validation?

Reply:
The `store` argument has been removed :-)

Reply:
Done in #1218. I'm not too familiar w/ `dbfs` so I'll leave artifact validation up to you all